File: agents/flight-openai/app/a2a/agent_card.py
from __future__ import annotations

from .models import AgentCapabilities, AgentCard, AgentSkill


def build_agent_card(public_url: str) -> AgentCard:
    return AgentCard(
        name="flight-agent",
        description="Searches and ranks flight options for a trip. Only handles flights — never hotels, activities or budget.",
        version="0.1.0",
        url=public_url,
        capabilities=AgentCapabilities(streaming=False, push_notifications=False),
        skills=[
            AgentSkill(
                id="search_flights",
                name="Search Flights",
                description="Searches for flights based on origin, destination, dates and traveler count via MCP Flight Search, returns up to 5 ranked options.",
                tags=["flight", "specialist"],
            )
        ],
    )

File: agents/hotel-langgraph/src/a2a/agentCard.ts
import type { AgentCard } from "./models.js";

export function buildAgentCard(publicUrl: string): AgentCard {
  return {
    name: "hotel-agent",
    description:
      "Searches, filters and ranks hotel options for a trip. Only handles hotels — never flights, activities or budget.",
    version: "0.1.0",
    url: publicUrl,
    protocol_version: "0.3",
    capabilities: { streaming: false, push_notifications: false },
    skills: [
      {
        id: "search_hotels",
        name: "Search Hotels",
        description:
          "Searches for hotels based on destination, dates, guest count, budget, location and preferences via MCP Hotel Search, returns ranked options.",
        tags: ["hotel", "specialist"],
      },
    ],
    default_input_modes: ["text/plain", "application/json"],
    default_output_modes: ["text/plain", "application/json"],
  };
}
